drivers/atwinc15x0/include/atwinc15x0_internal.h

```diff
@@ -34,6 +37,136 @@ extern "C" {
  */
 extern atwinc15x0_t *atwinc15x0;
 
+/**
+ * @brief   Flag to indicate that the transceiver is scanning at the moment
+ *          and is thus not able to start a new scan concurrently
+ */
+#define ATWINC15X0_FLAG_SCANNING            (((atwinc15x0_flags_t)1) << 0)
+
+/**
+ * @brief   Flag to indicate that the transceiver is trying to connect to an AP
+ *          at the moment and is thus not able to process another connect
+ *          request concurrently
+ */
+#define ATWINC1510_FLAG_CONNECTING          (((atwinc15x0_flags_t)1) << 1)
+
+/**
+ * @brief   Flag to indicate that the transceiver is connected to an AP
+ */
+#define ATWINC15X0_FLAG_CONNECTED           (((atwinc15x0_flags_t)1) << 2)
```

Comment:
Can those be set at the same time?
From a glance it feels like distinct states like

```C
typedef enum {
    ATWINC1510_STATE_DISCONNECTED,
    ATWINC1510_STATE_DISCONNECTED_SCANNING,
    ATWINC1510_STATE_CONNECTING,
    ATWINC1510_STATE_CONNECTED,
    ATWINC1510_STATE_CONNECTED_SCANNING,
} atwinc15x0_state_t;
``` 

would be easier to handler.

Reply:
Since
```C
    if (IS_USED(MODULE_ATWINC15X0_STATIC_CONNECT)) {
        /* try to connect and return */
        res = _atwinc15x0_static_connect();
    }
    if (IS_USED(MODULE_ATWINC15X0_DYNAMIC_SCAN)) {
        wifi_scan_request_t request = WIFI_SCAN_REQUEST_INITIALIZER(NETOPT_SCAN_REQ_ALL_CH,
                                                                    _wifi_scan_cb, 0);
        _atwinc15x0_scan(&request);
    }
```
did not work when both modules were enabled, `ATWINC1510_STATE_CONNECTING_SCANNING` would not be a valid state, so I think your state analysis is right. Maybe it requires one more `is_connected` check to distinguish between  `ATWINC1510_STATE_DISCONNECTED_SCANNING` and `ATWINC1510_STATE_CONNECTED_SCANNING`, but I can try to reformulate it with states.

Reply:
I am trying to turn `netopt_state_t state` into `atwinc15x0_state_t state` in `atwinc15x0_t`. It is a bit more effort than I thought, to think about what operation can be performed in which state.

Reply:
I must still test a bit with the state handling